specifyweb/frontend/js_src/lib/components/Toolbar/TreeRepair.tsx

```diff
@@ -92,16 +93,33 @@ export function TreeSelectDialog({
                   : hasTreeAccess(treeName, 'read')
               )
               .map((treeName) => {
-                const treeDefinition = treeRanks[treeName]?.definition as
-                  | SpecifyResource<TaxonTreeDef>
-                  | undefined;
+                const cachedDefinitionId = getCache(
+                  'tree',
+                  `definition${treeName}`
+                );
+
+                const treeDefinition = deserializeResource(
+                  getTreeDefinitions(
+                    treeName,
+                    cachedDefinitionId === undefined
+                      ? undefined
+                      : {
+                          id: cachedDefinitionId,
+                        }
+                  )[0].definition
+                );
```

Comment:
doing deserializeResource on each render may be a bit wasteful
could you put lines 86-110 into a useMemo?